### DISCUSSION TITLE
rlgl questions

### DISCUSSION THREAD

**Discussion by Germ210** [raysan5/raylib#4246] [Q&A] [answered]
Hello, I have a few questions about rlgl that I don't know where to find the answer to 
1. Is there any documentation to read? 
2. What dependencies do I need to build it?
3. How can I set it up with ANGLE and opengles 2.0?
4. Is it like java (assuming you I use a cross platform windowing library like glfw), where it's write once, run anywhere (with ANGLE)?
5. How easy is it and what are your opinions on it for those who used before?
6. What would I need to make a window resizable and work with ANGLE without any setup for each individual window?

Sorry if I bothered you

**Comment by Peter0x44** [comment, marked as answer]
> Is there any documentation to read?

rlgl.h itself

> What dependencies do I need to build it?

rlgl still needs an opengl extension loader, it uses glad. It is included in the repo.

> How can I set it up with ANGLE and opengles 2.0?

There are other discussions around that already. I dumped some info in this comment:
https://github.com/raysan5/raylib/discussions/2998#discussioncomment-6066227

> Is it like java (assuming you I use a cross platform windowing library like glfw), where it's write once, run anywhere (with ANGLE)?

You can "write once run anywhere", for the most part. But you cannot "compile once run anywhere" like Java.

> How easy is it and what are your opinions on it for those who used before?

Opinions on what? I think this is subjective and you should make up your own mind.

> What would I need to make a window resizable and work with ANGLE without any setup for each individual window?

rlgl doesn't create windows. You need to write code for that, or use an existing library like glfw or sdl.
Or, just "use raylib", which has that code. I'm not entirely sure why you're asking about rlgl specifically.
If you really insist on using rlgl "standalone", check this example:
https://github.com/raysan5/raylib/blob/master/examples/others/rlgl_standalone.c
But, I don't recommend that, unless you really know *why*.

**Reply by Germ210** [reply]
Yeah, I am reading the header file right now, along with comparing it to opengl 1.1
